src/main/java/net/minestom/server/network/packet/server/play/ChatMessagePacket.java

```diff
@@ -35,9 +81,29 @@ public int getId() {
         return ServerPacketIdentifier.CHAT_MESSAGE;
     }
 
+    /**
+     * @deprecated Use {@link MessageType}
+     */
+    @Deprecated
     public enum Position {
-        CHAT,
-        SYSTEM_MESSAGE,
-        GAME_INFO
+        CHAT(MessageType.CHAT),
+        SYSTEM_MESSAGE(MessageType.SYSTEM),
+        GAME_INFO(null);
+
+        private final MessageType messageType;
+
+        Position(MessageType messageType) {
+            this.messageType = messageType;
+        }
+
+        /**
+         * Gets this position as an Adventure message type. Note this will return
+         * {@code null} for {@link #GAME_INFO} as it is preferred to use
+         * {@link TitlePacket} due to <a href="https://bugs.mojang.com/browse/MC-119145">MC-119145</a>.
+         * @return the message type
+         */
+        public @Nullable MessageType asMessageType() {
+            return this.messageType;
+        }
```

Comment:
The MC bug is marked as resolved, is the null return still necessary?

Reply:
Well Adventure doesn't support `sendMessage` with a `MessageType` that indicates the message should be sent in the game info position. Following the reasons mentioned below, this `MessageType` enum should actually be switched to an int describing the message type. I'll wait for your response about the switch from enum to int.

Reply:
Resolved as per other discussions.